project/src/main/java/com/google/autograder/data/Database.java

```diff
@@ -119,9 +132,10 @@ public String getAllAssignmentsJSON() {
 
   public String getAllQuestionsJSON(String key) {
     // query all questions with this key at assignment_id key
-    if (key != null) {
+    try {
```

Comment:
try-catch blocks should be as narrow scope as possible, so that it's clear what lines of code can throw an exception

Reply:
makes sense!